test/test.c

```diff
@@ -31470,6 +31470,133 @@ void test_KeyValueKeys(void)
     JS_TEARDOWN;
 }
 
+void test_KeyValueKeysWithFilters(void)
+{
+    natsStatus          s;
+    kvStore             *kv = NULL;
+    kvKeysList          l;
+    kvConfig            kvc;
+    const char **defaultSubject = (const char *[]){">"};
+    l.Count = 0;
+    l.Keys = NULL;
+
+    JS_SETUP(2, 10, 14);
+
+    test("Create KV: ");
+    kvConfig_Init(&kvc);
+    kvc.Bucket = "KVSF";
+    kvc.History = 2;
+    s = js_CreateKeyValue(&kv, js, &kvc);
+    testCond(s == NATS_OK);
+
+    test("Populate: ");
+    s = kvStore_PutString(NULL, kv, "a.b", "a.b");
+    IFOK(s, kvStore_PutString(NULL, kv, "a.d", "a.d"));
+    IFOK(s, kvStore_PutString(NULL, kv, "c.d", "c.d"));
+    IFOK(s, kvStore_PutString(NULL, kv, "e.f", "e.f"));
+    IFOK(s, kvStore_PutString(NULL, kv, "e.a.f", "e.a.f"));
+    testCond(s == NATS_OK);
+
+    test("Get keys with filters (bad args): List is NULL");
+    s = kvStore_KeysWithFilters(NULL, kv, NULL, defaultSubject, 1);
+    testCond(s == NATS_INVALID_ARG);
+    nats_clearLastError();
+
+    test("Get keys with filters (bad args): filters is NULL");
+    s = kvStore_KeysWithFilters(&l, kv, NULL, NULL,0);
+    testCond((s == NATS_INVALID_ARG) && (l.Keys == NULL) && (l.Count == 0));
+    nats_clearLastError();
+
+    test("Get keys with filters (bad args): numFilters is 0");
+    s = kvStore_KeysWithFilters(&l, kv, NULL, defaultSubject,0);
+    testCond((s == NATS_INVALID_ARG) && (l.Keys == NULL) && (l.Count == 0));
+    nats_clearLastError();
+
+    test("Get keys with filters (bad args): kv is NULL");
```

Comment:
Before this test, you could add:
```
test("Get keys with filters (bad args): numFilters is <0");
s = kvStore_KeysWithFilters(&l, kv, NULL, defaultSubject, -10);
testCond((s == NATS_INVALID_ARG) && (l.Keys == NULL) && (l.Count == 0));
nats_clearLastError();

test("Get keys with filters (bad args): empty string");
const char **filter0 = (const char *[]){"a.*", "", "b.*"};
s = kvStore_KeysWithFilters(&l, kv, NULL, filter0, 3);
testCond((s == NATS_INVALID_ARG) && (l.Keys == NULL) && (l.Count == 0));
nats_clearLastError();
```
Note that you don't have to specifically check in your code for empty filter strings since this is done by WatchMulti. The only check you do is that it is not NULL or <=0, because this is what you need to differentiate which call `_kvStore_Keys()` is going to make. But there is no need to "redo" the checks that are done in the call stack.

Reply:
Thanks for the suggestion. will add these tests as well.